### DISCUSSION TITLE
vgui_russian and serverbrowser.txt

### DISCUSSION THREAD

**Discussion by Duke-M-commits** [ReactiveDrop/reactivedrop_translations#107] [Q&A] [answered]
I realized there are some files in the repo for which I do not have an english original file.
serverbrowser_vietnamese.txt - it is there for long now
vgui_russian.txt - this one's pretty new

I found both files as a german version in my steam installation at `C:\Program Files (x86)\Steam\servers\serverbrowser_german.txt`.
My server browser only switches to german in-game when I change the Steam clients language.
Does this mean I can override this behaviour when I drop a german version into the games translation?

I also realized the line-count of the russian file doesn't match up.

So now I'm wondering ... how do I translate the server browser?
Can I add a small file which just overrides some of the values and the rest is pulled from the steam installation?
Or do I have to provide a full translation?
I also saw keys in the russian translation which aren't in the steam ones, like `KBEditorCommand`.

Please help :)
@geotavros
@Blueberryy
@BenLubar

**Comment by Blueberryy** [comment]
i got some of those russian files from CSGO as it's more updated so yeah it may include some not existing strings.

**Comment by BenLubar** [comment, marked as answer]
Removing serverbrowser.dll and replacing it with something we fully control is on my long term TODO list. There's a bunch of stuff the server browser does that I don't like (non-controller-accessible UI, the weird corrupted tags, etc.) The server browser UI is almost a 1:1 match for the [ISteamMatchmakingServers](https://partner.steamgames.com/doc/api/ISteamMatchmakingServers) API, so it's just a matter of designing and implementing a new UI.

Currently the translations shipped with the game are in the `credits`, `addons`, and `resource` folders. The `achievements`, `community`, `rich_presence`, and `store_page` folders are used by out-of-game things, mostly uploaded manually on the partner website.

I don't think `platform` from this repo is currently used at all (the build script only uses the internal `swarm_for_rd` and `reactivedrop_dlls` sources to create the `platform` folder in the released game, not the `reactivedropgit` source which is derived from `reactivedrop_public_src`.)
